// vue-drag-handle/drag-handle.js
;(function(){
module.exports = {
  mixins: [require("vue-mixins/style"), require("vue-mixins/class")],
  props: {
    id: {
      type: String
    },
    style: {
      "default": function() {}
    },
    "class": {
      "default": function() {}
    },
    factor: {
      type: Number,
      "default": 2
    },
    maxLeft: {
      type: Number,
      "default": 0
    },
    maxRight: {
      type: Number,
      "default": 0
    },
    offset: {
      type: Number,
      "default": 0
    },
    zIndex: {
      type: Number,
      "default": 1002
    },
    disabled: {
      type: Boolean,
      "default": false
    }
  },
  computed: {
    mergeStyle: function() {
      return {
        height: "100%",
        position: "absolute",
        top: "0",
        zIndex: this.zIndex
      };
    }
  },
  data: function() {
    return {
      atMax: false,
      pos: null
    };
  },
  methods: {
    click: function(e) {
      if (!((this.pos != null) && this.pos.x === e.x && this.pos.y === e.y)) {
        return this.$emit("clean-click", e);
      }
    },
    onPan: function(e) {
      var dX;
      if (e.type === "pan") {
        e.srcEvent.preventDefault();
        dX = e.deltaX * this.factor;
        this.pos = null;
        if (e.isFinal) {
          this.pos = {
            x: e.srcEvent.x,
            y: e.srcEvent.y
          };
          if (this.maxRight > 0 && dX >= this.maxRight) {
            this.$emit("right");
            return this.$emit("max", "right");
          } else if (this.maxLeft > 0 && dX <= -this.maxLeft) {
            this.$emit("left");
            return this.$emit("max", "left");
          } else {
            return this.$emit("aborted");
          }
        } else if (this.maxRight > 0 && dX >= 0) {
          if (dX <= this.maxRight) {
            this.$emit("move", dX + this.offset);
            return this.atMax = false;
          } else if (!this.atMax) {
            this.$emit("move", this.maxRight + this.offset);
            return this.atMax = true;
          }
        } else if (this.maxLeft > 0 && dX <= 0) {
          if (dX >= -this.maxLeft) {
            this.$emit("move", dX + this.offset);
            return this.atMax = false;
          } else if (!this.atMax) {
            this.$emit("move", -this.maxLeft + this.offset);
            return this.atMax = true;
          }
        }
      }
    }
  }
};

})()
if (module.exports.__esModule) module.exports = module.exports.default
var __vue__options__ = (typeof module.exports === "function"? module.exports.options: module.exports)
__vue__options__.render = function(){with(this){return (!disabled)?_h('div',{directives:[{name:"touch",value:(onPan),expression:"onPan",arg:"pan"}],staticClass:"drag-handle",class:computedClass,style:(computedStyle),attrs:{"id":id},on:{"click":click}}):_e()}}
__vue__options__.staticRenderFns = []
